google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -52,23 +64,89 @@ def __init__(
         """Retrieves an existing managed dataset given a dataset name or ID.
 
         Args:
-            dataset_name (str):
+            dataset_name: (str)
                 Required. A fully-qualified dataset resource name or dataset ID.
                 Example: "projects/123/locations/us-central1/datasets/456" or
                 "456" when project and location are initialized or passed.
-            project (str):
+            project: (str) = None
                 Optional project to retrieve dataset from. If not set, project
                 set in aiplatform.init will be used.
-            location (str):
+            location: (str) = None
                 Optional location to retrieve dataset from. If not set, location
                 set in aiplatform.init will be used.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials: Optional[auth_credentials.Credentials] = None,
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the dataset type of the retrieved dataset resource is
+            not supported by the 'Dataset' class.
         """
 
         super().__init__(project=project, location=location, credentials=credentials)
         self._gca_resource = self._get_gca_resource(resource_name=dataset_name)
+        self._validate_resource_metadata_schema_uri()
+
+    @property
+    def resource_metadata_schema_uri(self) -> str:
+        """The metadata schema uri of this dataset resource."""
+        return self._gca_resource.metadata_schema_uri
+
+    def _validate_resource_metadata_schema_uri(self) -> bool:
```

Comment:
feels strange that this can never return false. Perhaps return false and raise the error in the caller instead?

Reply:
As I understand, this function is to make sure that `ImageDataset.init` should only ever return an Dataset with a matching `_gca_resource.metadata_schema_uri`, in that case, I think this validation logic should live in `ImageDataset` and not in the Dataset class. 

Dataset doesn't need to know about subclass-specific logic.

Reply:
If you want to reuse code, perhaps a helper function or a separate validation class would be more appropriate than putting this into the Dataset class.

Reply:
The validation is to valid when init (retrieving) an existing managed dataset, both image and tabular metadata_schema_uri validation share the same logic, that's why it was put under Dataset class, and subclasses can override the _support_metadata_schema_uris. I will explore a validation class for schemas, the import schema also need validations.

Reply:
I understand that both subclasses need it, but a Dataclass instance shouldn't need to know about that, hence the logic may better fit in the subclasses instead of the shared superclass.

Reply:
note: Discussion continued offline.